flask/src/main/java/net/corda/gradle/flask/JavaAgent.java

```diff
@@ -0,0 +1,64 @@
+package net.corda.gradle.flask;
+
+import org.gradle.api.Named;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.plugins.ExtensionContainer;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.PathSensitive;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import static org.gradle.api.tasks.PathSensitivity.RELATIVE;
+
+public class JavaAgent implements Named {
+    private final String name;
+    private final RegularFileProperty jar;
+    private final Property<String> args;
+
+    @Inject
+    public JavaAgent(String name, ObjectFactory objects) {
+        this.name = name;
+        this.jar = objects.fileProperty();
+        this.args = objects.property(String.class);
+    }
+
+    @Override
+    @Nonnull
+    @Input
+    public String getName() {
+        return name;
+    }
+
+    @InputFile
+    @PathSensitive(RELATIVE)
+    public Provider<RegularFile> getJar() {
+        return jar;
+    }
+
+    public void setJar(Provider<RegularFile> jar) {
+        this.jar.set(jar);
+    }
+
+    public void setJar(RegularFile jar) {
+        this.jar.set(jar);
+    }
+
+    @Input
```

Comment:
This should be
```java
@Optional
@Input
```
as otherwise Gradle will complain at runtime if this agent has no args.